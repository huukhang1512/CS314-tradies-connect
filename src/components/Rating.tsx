import React, { useState } from "react";
import { Box, Stack, Text } from "@chakra-ui/react";
import { GiRoundStar } from "react-icons/gi";

export type RatingProps = {
  size?: number;
  onChange: (rating: number) => void;
  value: number;
  maxRating: number;
  name: string
}

const Rating = ({ size, onChange, value, maxRating }: RatingProps) => {
  const [rating, setRating] = useState(value);

  const onClick = (idx) => {
    if (!isNaN(idx)) {
      setRating(idx);
      onChange(idx)
    }
  };

  const RatingIcon = ({ fill }) => {
    return (
      <GiRoundStar
        size={"3em"}
        color={fill ? "#FFDB5E" : "#D8DAE2"}
        onClick={onClick}
      />
    );
  };

  const RatingButton = ({ idx, fill }) => {
    return (
      <Box
        as="button"
        aria-label={`Rate ${idx}`}
        height={`${size}px`}
        width={`${size}px`}
        mx={1}
        onClick={() => onClick(idx)}
        _focus={{ outline: 0 }}
      >
        <RatingIcon fill={fill} />
      </Box>
    );
  };

  return (
    <Stack isInline mt={3} justify="center">
      {[...Array(maxRating)].map((_, i) => (
        <RatingButton key={i + 1} idx={i + 1} fill={i < rating} />
      ))}
    </Stack>
  );
};

export default Rating;
