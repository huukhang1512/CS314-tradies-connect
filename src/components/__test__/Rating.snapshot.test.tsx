/**
 * @jest-environment jsdom
 */
import React from "react";
import { render } from "@testing-library/react";
import Rating from "../Rating";

describe("<Rating/> test", () => {
  it("Matches DOM Snapshot", () => {
    const { container } = render(
      <Rating
        maxRating={5}
        name="test"
        size={3}
        value={3}
        onChange={() => console.log()}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
