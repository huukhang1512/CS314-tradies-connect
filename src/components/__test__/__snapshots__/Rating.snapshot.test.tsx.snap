// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Rating/> test Matches DOM Snapshot 1`] = `
<div>
  <div
    class="chakra-stack css-17ta7it"
  >
    <button
      aria-label="Rate 1"
      class="css-oeh5ny"
    >
      <svg
        color="#FFDB5E"
        fill="currentColor"
        height="3px"
        stroke="currentColor"
        stroke-width="0"
        style="color: rgb(255, 219, 94);"
        viewBox="0 0 512 512"
        width="3px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 38.013c-22.458 0-66.472 110.3-84.64 123.502-18.17 13.2-136.674 20.975-143.614 42.334-6.94 21.358 84.362 97.303 91.302 118.662 6.94 21.36-22.286 136.465-4.116 149.665 18.17 13.2 118.61-50.164 141.068-50.164 22.458 0 122.9 63.365 141.068 50.164 18.17-13.2-11.056-128.306-4.116-149.665 6.94-21.36 98.242-97.304 91.302-118.663-6.94-21.36-125.444-29.134-143.613-42.335-18.168-13.2-62.182-123.502-84.64-123.502z"
        />
      </svg>
    </button>
    <button
      aria-label="Rate 2"
      class="css-oeh5ny"
    >
      <svg
        color="#FFDB5E"
        fill="currentColor"
        height="3px"
        stroke="currentColor"
        stroke-width="0"
        style="color: rgb(255, 219, 94);"
        viewBox="0 0 512 512"
        width="3px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 38.013c-22.458 0-66.472 110.3-84.64 123.502-18.17 13.2-136.674 20.975-143.614 42.334-6.94 21.358 84.362 97.303 91.302 118.662 6.94 21.36-22.286 136.465-4.116 149.665 18.17 13.2 118.61-50.164 141.068-50.164 22.458 0 122.9 63.365 141.068 50.164 18.17-13.2-11.056-128.306-4.116-149.665 6.94-21.36 98.242-97.304 91.302-118.663-6.94-21.36-125.444-29.134-143.613-42.335-18.168-13.2-62.182-123.502-84.64-123.502z"
        />
      </svg>
    </button>
    <button
      aria-label="Rate 3"
      class="css-oeh5ny"
    >
      <svg
        color="#FFDB5E"
        fill="currentColor"
        height="3px"
        stroke="currentColor"
        stroke-width="0"
        style="color: rgb(255, 219, 94);"
        viewBox="0 0 512 512"
        width="3px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 38.013c-22.458 0-66.472 110.3-84.64 123.502-18.17 13.2-136.674 20.975-143.614 42.334-6.94 21.358 84.362 97.303 91.302 118.662 6.94 21.36-22.286 136.465-4.116 149.665 18.17 13.2 118.61-50.164 141.068-50.164 22.458 0 122.9 63.365 141.068 50.164 18.17-13.2-11.056-128.306-4.116-149.665 6.94-21.36 98.242-97.304 91.302-118.663-6.94-21.36-125.444-29.134-143.613-42.335-18.168-13.2-62.182-123.502-84.64-123.502z"
        />
      </svg>
    </button>
    <button
      aria-label="Rate 4"
      class="css-oeh5ny"
    >
      <svg
        color="#D8DAE2"
        fill="currentColor"
        height="3px"
        stroke="currentColor"
        stroke-width="0"
        style="color: rgb(216, 218, 226);"
        viewBox="0 0 512 512"
        width="3px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 38.013c-22.458 0-66.472 110.3-84.64 123.502-18.17 13.2-136.674 20.975-143.614 42.334-6.94 21.358 84.362 97.303 91.302 118.662 6.94 21.36-22.286 136.465-4.116 149.665 18.17 13.2 118.61-50.164 141.068-50.164 22.458 0 122.9 63.365 141.068 50.164 18.17-13.2-11.056-128.306-4.116-149.665 6.94-21.36 98.242-97.304 91.302-118.663-6.94-21.36-125.444-29.134-143.613-42.335-18.168-13.2-62.182-123.502-84.64-123.502z"
        />
      </svg>
    </button>
    <button
      aria-label="Rate 5"
      class="css-oeh5ny"
    >
      <svg
        color="#D8DAE2"
        fill="currentColor"
        height="3px"
        stroke="currentColor"
        stroke-width="0"
        style="color: rgb(216, 218, 226);"
        viewBox="0 0 512 512"
        width="3px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 38.013c-22.458 0-66.472 110.3-84.64 123.502-18.17 13.2-136.674 20.975-143.614 42.334-6.94 21.358 84.362 97.303 91.302 118.662 6.94 21.36-22.286 136.465-4.116 149.665 18.17 13.2 118.61-50.164 141.068-50.164 22.458 0 122.9 63.365 141.068 50.164 18.17-13.2-11.056-128.306-4.116-149.665 6.94-21.36 98.242-97.304 91.302-118.663-6.94-21.36-125.444-29.134-143.613-42.335-18.168-13.2-62.182-123.502-84.64-123.502z"
        />
      </svg>
    </button>
  </div>
</div>
`;
